EnvironmentSimulator/Modules/ScenarioEngine/OSCTypeDefs/OSCPrivateAction.cpp

```diff
@@ -312,6 +312,14 @@ void LatLaneChangeAction::Step(double dt, double)
 	double factor;
 	double angle = 0;
 
+	// Detect whether t switched sign due to end-to-end/start-to-start succession of roads
+	if (fabs(t_old + object_->pos_.GetT()) < SMALL_NUMBER && fabs(t_old) > SMALL_NUMBER)
```

Comment:
I guess that the purpose of the check `fabs(t_old) > SMALL_NUMBER` is to avoid false cases close to ref line (where the sum may be less than SMALL_NUMBER in spite of same sign). 

This will probably work for 99.999% of all cases. However, if it happens that the road switch happens exactly when t is close to 0, then it might not be detected.

So, to cover all cases, what do you think of making this slight change to the statement:
`if (SIGN(object_->pos_.GetT()) != SIGN(t_old) && fabs(t_old + object_->pos_.GetT()) < SMALL_NUMBER)`
?

Reply:
Yes, that was my intention - but I agree, your suggestion is much better and should capture 100% instead of 99%..

Reply:
Great. Would you like to do the change?

Reply:
Sure, I can do that. :)
Thanks for your feedback!

Reply:
Perfect!